include/volume.h

```diff
@@ -307,4 +359,160 @@ NT volume(T &P,
+        if(print) std::cout << "Rounding time = " << tstop1 - tstart1 << std::endl;
+        round_value=res_round.first;
+        std::pair<Point,NT> res=solveLP(P);
+        c=res.first; radius=res.second;
+    }
+    if(var.ball_walk){
+        var.che_rad = radius;
+    }
+
+    //1. Move chebychev center to origin and apply the same shifting to the polytope
+    int m=P.num_of_hyperplanes();
+    Eigen::MatrixXd A(m,n);
+    Eigen::VectorXd b(m);
+    Eigen::VectorXd c_e(n);
+    for(int i=0; i<n; i++){
+        c_e(i)=c[i];
+    }
+    for(int i=0; i<m; ++i){
+        b(i) = P.get_coeff(i,0);
+        for(int j=1; j<n+1; ++j){
+            A(i,j-1) = P.get_coeff(i,j);
+        }
+    }
+    //Shift polytope
+    b = b - A*c_e;
+    // Write changesto the polytope!
+    for(int i=0; i<m; ++i){
+        P.put_coeff(i,0,b(i));
+        for(int j=1; j<n+1; ++j){
+            P.put_coeff(i,j,A(i,j-1));
+        }
+    }
+
+    std::vector<NT> a_vals;
+    NT ratio = var.ratio;
+    NT C = var.C;
+
+    if(print) std::cout<<"\n\nComputing annealing...\n"<<std::endl;
+    int N = var.N;
+    double tstart2 = (double)clock()/(double)CLOCKS_PER_SEC;
+    get_annealing_schedule(P, a_vals, error, radius, ratio, C, frac, N, var);
+    double tstop2 = (double)clock()/(double)CLOCKS_PER_SEC;
+    if(print) std::cout<<"All the variances of schedule_Sannealing computed in = "<<tstop2-tstart2<<" sec"<<std::endl;
+    int mm = a_vals.size();
+    if(print){
+        for(int i=0; i<mm; i++){
+            std::cout<<"a_"<<i<<" = "<<a_vals[i]<<" ";
+        }
+        std::cout<<"\n"<<std::endl;
+    }
+    std::vector<NT> fn(mm,0), its(mm,0), lamdas(m,0);
+    int W = var.W;
+    std::vector<NT> last_W2(W,0);
+    vol=std::pow(M_PI/a_vals[0], (NT(n))/2.0)*std::abs(round_value);
+    if(print) std::cout<<"volume of the first gaussian = "<<vol<<"\n"<<std::endl;
+    Point p(n);
+    std::pair<int,NT> res;
+    Point p_prev=p;
+    int coord_prev=-1;
+
+    if(print) std::cout<<"computing ratios..\n"<<std::endl;
+    for(int i=0; i<mm-1; i++){
+        //initialize convergence test
+        curr_eps = error/std::sqrt((NT(mm)));
+        done=false;
+        min_val=-std::pow(10.0,10.0);
+        max_val=-min_val;
+        min_index=W-1;
+        max_index=W-1;
+        index = 0;
+        min_steps=0;
+        std::vector<NT> last_W=last_W2;
+        n_threads=1;
+
+        while(!done || its[i]<min_steps){
+
+            gaussian_next_point(P,p,p_prev,coord_prev,var.walk_steps,a_vals[i],lamdas,var);
+
+            if(!P.is_in(p)){
+                std::cout<<"point not in P\n";
+                exit(-1);
+            }
+            its[i] += 1.0;
+            fn[i] += eval_exp(p,a_vals[i+1]) / eval_exp(p,a_vals[i]);
+            val = fn[i]/its[i];
+
+            last_W[index] = val;
+            if(val<=min_val){
+                min_val = val;
+                min_index = index;
+            }else if(min_index==index){
+                res=min_vec(last_W);
```

Comment:
why `min_vec` is defined in `volume.h` if only used here? Also these functions are implemented in the standard library, see `min_element`

Reply:
Done. I forgot to change this after the first implementation. We use iterator from min_element and max_element now. It seems that it gives to the implementation a speed up by almost 30%!